controller/volume_rebuilding_controller.go

```diff
@@ -190,14 +252,124 @@ func (vbc *VolumeRebuildingController) reconcile(volName string) (err error) {
 			return
 		}
 	}()
+	existingVol := vol.DeepCopy()
+	defer func() {
+		if err != nil {
+			return
+		}
+		if reflect.DeepEqual(existingVol.Spec, vol.Spec) {
+			return
+		}
+
+		if _, err = vbc.ds.UpdateVolume(vol); err != nil {
+			return
+		}
+	}()
 
 	rebuildingAttachmentTicketID := longhorn.GetAttachmentTicketID(longhorn.AttacherTypeVolumeRebuildingController, volName)
+	csiTicketExists := isVATicketExists(va, longhorn.AttacherTypeCSIAttacher)
+	isOfflineRebuildEnabled, err := vbc.isVolumeOfflineRebuildEnabled(vol.Spec.OfflineRebuild)
+	if err != nil {
+		return err
+	}
 
-	delete(va.Spec.AttachmentTickets, rebuildingAttachmentTicketID)
+	if isOfflineRebuildEnabled && vol.Status.Robustness != longhorn.VolumeRobustnessFaulted && !csiTicketExists {
```

Comment:
What about if other important tickets such as longhorn-ui, restore controller, clone controller exist? CSI ticket is not only the highest priority ticket. Maybe we can check if there are no other ticket here?

Reply:
Yes, it's a good idea.
Modified to be that higher priority VA ticket can preempt the volume rebuild VA ticket.
Thanks